mobilecoind/src/service.rs

```diff
@@ -4184,6 +4292,226 @@ mod test {
+            .next()
+            .unwrap();
+        let offered_value = offered_input.value;
+
+        // Generate a swap.
+        let generate_swap_response = {
+            let mut request = api::GenerateSwapRequest::new();
+            request.set_sender_monitor_id(originator_monitor_id.to_vec());
+            request.set_change_subaddress(0);
+            request.set_input(offered_input);
+            request.set_allow_partial_fill(true);
+            request.set_counter_value(123);
+            request.set_counter_token_id(1);
+            request.set_minimum_fill_value(10);
+            client.generate_swap(&request).unwrap()
+        };
+
+        // Now we will try to build a transaction that incorporates the swap.
+        // The counterparty needs to supply eUSD.
+        // Get list of unspent tx outs
+        let utxos = mobilecoind_db
+            .get_utxos_for_subaddress(&counterparty_monitor_id, 0)
+            .unwrap();
+        assert!(!utxos.is_empty());
+
+        // We will try to take half of the offered input.
+        let mut sci_for_tx = api::SciForTx::new();
+        sci_for_tx.set_sci(generate_swap_response.get_sci().clone());
+        sci_for_tx.set_partial_fill_value(offered_value / 4);
+
+        // Try to add the swap to a mixed transaction
+        let mut request = api::GenerateMixedTxRequest::new();
+        request.set_sender_monitor_id(counterparty_monitor_id.to_vec());
+        request.set_change_subaddress(0);
+        // Only token id 1 is needed to fulfill the other side of the sci
+        request.set_input_list(
+            utxos
+                .iter()
+                .filter(|utxo| utxo.token_id == 1)
+                .map(Into::into)
+                .collect(),
+        );
+        request.set_scis(vec![sci_for_tx].into());
+        let generate_mixed_tx_response = client.generate_mixed_tx(&request).unwrap();
+
+        assert_eq!(
+            generate_mixed_tx_response
+                .get_tx_proposal()
+                .get_scis()
+                .len(),
+            1
+        );
+
+        let tx = Tx::try_from(generate_mixed_tx_response.get_tx_proposal().get_tx()).unwrap();
+
+        assert_eq!(tx.prefix.outputs.len(), 4);
+
+        let mut found_counterparty_output = false;
+        let mut found_counterparty_change = false;
+        let mut found_originator_output = false;
+        let mut found_originator_change = false;
+        for output in tx.prefix.outputs.iter() {
+            if let Ok((amount, _scalar)) =
+                output.view_key_match(swap_counterparty.view_private_key())
+            {
+                if amount.token_id == Mob::ID {
+                    assert_eq!(amount.value, offered_value / 4 - Mob::MINIMUM_FEE);
+                    found_counterparty_output = true;
+                } else {
+                    assert_eq!(*amount.token_id, 1);
+                    // It's not very easy to figure out what the value of this change output should
```

Comment:
Isn't it going to be the offered amount minus claimed amount (which is hardcoded below)?

Reply:
I think what happens is, we need to figure out what is the value of the token id 1 input that `swap_counterparty` will select. Then we can figure out what the size of the token id 1 change output should be for them.

I think I could likely figure this out by filtering in the `utxo` set and getting the value of the first one, I can follow up with that.